### ISSUE TITLE
Implement NanosecondsSinceEpoch for MacOS X

### ISSUE THREAD

**Issue by markuspf** [gap-system/gap#750]
MacOS X seems to have a special API for getting clock times. Implement `NanosecondsSinceEpoch` using that.

**Comment by markuspf** [comment]
I think @ChrisJefferson had this sorted already

**Comment by ChrisJefferson** [comment]
For the purpose of Mac OS X, it's hard (almost impossible) to get NanosecondsSinceEpoch. Does anyone need specifically since Epoch, or just from some fixed point (in Mac's case, from application start)?

Trying to think of a good alternative name without Epoch.. Just Nanoseconds() ? (not sure)

**Comment by markuspf** [comment]
On Mon, Apr 18, 2016 at 05:38:49AM -0700, Christopher Jefferson wrote:
> For the purpose of Mac OS X, it's hard (almost impossible) to get
> NanosecondsSinceEpoch. Does anyone need specifically since Epoch, or just from
> some fixed point (in Mac's case, from application start)?
> 
> Trying to think of a good alternative name without Epoch.. Just Nanoseconds() ?
> (not sure)

"Epoch" does *not* refer to 1.1.1970 in this case, but just to a fixed point
in time. For example on dragonflybsd its since computer was booted. This is
also documented.

I actually looked this up and I had the same concerns about "Epoch", but then
ultimately decided that "Epoch" is the word to use.

Noone should ever rely on this number being "nanoseconds since 1.1.1970
00:00:00".

I found calling it "Nanoseconds" a bit too unspecific, hence I called it
MonotonicNanoseconds (which Max might not have liked). I just settled the
Bikeshed by using some name. Feel free to change it.
